pkg/artifactory/provider.go

```diff
@@ -110,12 +111,19 @@ func providerConfigure(d *schema.ResourceData, terraformVersion string) (interfa
 
 	var client *http.Client
 	details := auth.NewArtifactoryDetails()
+	u, err := url.ParseRequestURI(d.Get("url").(string))
+	if err != nil {
+		return nil, err
+	}
 
-	url := d.Get("url").(string)
-	if url[len(url)-1] != '/' {
-		url += "/"
+
+	if !strings.HasSuffix(u.String(), "/") {
+		u, err = url.ParseRequestURI(u.String() + "/")
```

Comment:
just curious, why url is being validated with "/" concatenated  again as it was already done above and nothing changed for that url in between?

Reply:
I am confused (and perhaps I did something wrong). The idea is:
if the url doesn't have a `/` on the end, add it. 
Is that not what this does?
Side note: I think I may have found a simpler way to normalize this

Reply:
// or / is treated the same in most scenarios but the above LGTM.

Reply:
RT is fussy - requires a trailing `/` or it will generate a 301 - and I don't know if the client is set to handle that - plus, even if it is, it will generate unnecessary chatter/latency in a large scale TF

Reply:
> I am confused (and perhaps I did something wrong). The idea is:
> if the url doesn't have a `/` on the end, add it.
> Is that not what this does?
> Side note: I think I may have found a simpler way to normalize this

its just that "ParseRequestURI" is applied over twice for the same string which i think was not needed, in the second part just concatenation would have been enough. not a big deal though